e2e/jail/jail_rpc_test.go

```diff
@@ -35,16 +35,46 @@ func (s *JailRPCTestSuite) SetupTest() {
 	s.NotNil(s.jail)
 }
 
+// TestJailRPCAsyncSend was written to catch race conditions with a weird error message
+// starting from `ReferenceError` as if otto vm were losing symbols.
+func (s *JailRPCTestSuite) TestJailRPCAsyncSend() {
```

Comment:
I think this test was removed (can't find commit message, should be in slack somewhere). The reason is that waiting for async tasks to complete is broken and causes goroutines leaks and panics in tests.

Reply:
Yeah, this test should be rewritten, at least. 

You're right that there is still a chance that running a task asynchronously, it will panic if a node gets stopped. I think `NodeManager` refactoring is needed to handle it properly: https://github.com/status-im/status-go/issues/403